model/task/task.go

```diff
@@ -794,10 +804,14 @@ func (t *Task) DependenciesMet(depCaches map[string]Task) (bool, error) {
 	}
 	// this is not exact, but depTask.FinishTime is not always set in time to use that
 	t.DependenciesMetTime = time.Now()
+	t.DisplayStatusCache = t.findDisplayStatus()
 	err = UpdateOne(
 		bson.M{IdKey: t.Id},
 		bson.M{
-			"$set": bson.M{DependenciesMetTimeKey: t.DependenciesMetTime},
+			"$set": bson.M{
+				DependenciesMetTimeKey: t.DependenciesMetTime,
+				DisplayStatusCacheKey:  t.DisplayStatusCache,
```

Comment:
Is there a particular reason we would update the display status cache here when it's just setting the dependency met time? The dependency met time doesn't seem to affect the display status, and if the dependencies were already marked as met previously, then the status would have been updated with the dependencies being met.